modules/features/referrals/src/main/kotlin/au/com/shiftyjelly/pocketcasts/referrals/ReferralCardAnimatedGradientView.kt

```diff
@@ -0,0 +1,155 @@
+        targetValue = positionVectors[1],
+        typeConverter = TwoWayConverter(
+            convertToVector = { position -> position },
+            convertFromVector = { vector ->
+                AnimationVector2D(vector.v1, vector.v2)
+            },
+        ),
+        animationSpec = infiniteRepeatable(
+            animation = keyframes {
+                durationMillis = animationDuration.toInt()
+                positionVectors.forEachIndexed { index, vector ->
+                    val timeStamp = index * (animationDuration.toInt() / positionVectors.size)
+                    vector.next() at timeStamp using LinearOutSlowInEasing
+                }
+            },
+            repeatMode = RepeatMode.Restart,
+        ),
+        label = "referral-card-background-animation",
+    )
+
+    BoxWithConstraints(
+        contentAlignment = Alignment.Center,
+        modifier = modifier
+            .background(Color.Black)
+            .clipToBounds(),
+    ) {
+        val density = LocalDensity.current
+        val circleSize = maxHeight
+        val circleSizePx = density.run { circleSize.toPx() }
+        val maxWidthPx = density.run { maxWidth.toPx() }
+        val maxHeightPx = density.run { maxHeight.toPx() }
+
+        val offsetX = (animatePosition.v1 * maxWidthPx) / 2f
+        val offsetY = (animatePosition.v2 * maxHeightPx) / 2f
+
+        GradientCircle(
+            circleSize = circleSize,
+            backgroundBrush = rainbowBrush(
+                start = Offset(0.12f * circleSizePx, 0f * circleSizePx),
+                end = Offset(0.89f * circleSizePx, 0.95f * circleSizePx),
+            ),
+            offset = IntOffset(offsetX.toInt(), offsetY.toInt()),
+        )
+
+        GradientCircle(
+            circleSize = circleSize,
+            backgroundBrush = rainbowBrush(
+                start = Offset(0.29f * circleSizePx, 0.19f * circleSizePx),
+                end = Offset(0.87f * circleSizePx, 1.18f * circleSizePx),
+            ),
+            offset = IntOffset(-offsetX.toInt(), -offsetY.toInt()),
+            rotation = 45f,
+        )
+    }
+}
+
+@Composable
+private fun GradientCircle(
+    circleSize: Dp,
+    offset: IntOffset,
+    backgroundBrush: Brush,
+    rotation: Float = 0f,
+) {
+    val modifier = Modifier
+        .offset { offset }
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+        Box(
+            modifier = modifier
+                .size(circleSize)
+                .background(
+                    brush = backgroundBrush,
+                    shape = CircleShape,
+                ),
+        )
+    } else {
+        Image(
+            painterResource(IR.drawable.blurred_rainbow_circle),
+            contentDescription = null,
+            contentScale = ContentScale.Crop,
+            modifier = modifier
+                .rotate(rotation)
+                .size(circleSize * 1.25f),
+        )
+    }
+}
+
+sealed class Position(val vector: AnimationVector2D) {
```

Comment:
I wonder if instead it could be turned into an `enum class` with a `fromVector()` function on the companion object and some other properties like `next()`? I think it could potentially clean up some stuff in the infinity transition and avoid declaration of things like `positionVectors` in favor of `Enum.entries`. I'm not sure if it will be cleaner though so feel free to ignore this.